esp/services/ws_workunits/ws_wudetails.cpp

```diff
@@ -0,0 +1,428 @@
+    unsigned __int64 maxTimestamp = 0;
+    IArrayOf<IEspWUResponseProperty> EspWUResponseProperties;
+    StatisticScopeType currentStatisticScopeType = SSTnone;
+
+    void buildAttribListToReturn(IConstWUPropertiesToReturn & propertiesToReturn);
+};
+
+WUDetailsVisitor::WUDetailsVisitor(IConstWUPropertyOptions & propertyOptions, IConstWUPropertiesToReturn & propertiesToReturn)
+{
+    includeName = propertyOptions.getIncludeName();
+    includeRawValue = propertyOptions.getIncludeRawValue();
+    includeFormatted = propertyOptions.getIncludeFormatted();
+    includeMeasure = propertyOptions.getIncludeMeasure();
+    includeCreator = propertyOptions.getIncludeCreator();
+    includeCreatorType = propertyOptions.getIncludeCreatorType();
+
+    buildAttribListToReturn(propertiesToReturn);
+}
+
+void WUDetailsVisitor::noteStatistic(StatisticKind kind, unsigned __int64 value, IConstWUStatistic & extra)
+{
+    if (extraStatisticsRequested)
+    {
+        // If the statistic is not in the standard return statistic list,
+        // then check if it's in the additional property list
+        if (returnStatisticListSpecified && !returnStatisticList->test(kind))
+        {
+            if (additionalStatsForCurScope==nullptr ||
+                additionalStatsForCurScope->find(kind)==additionalStatsForCurScope->end())
+            {
+                return;
+            }
+        }
+    }
+
+    Owned<IEspWUResponseProperty> EspWUResponseProperty = createWUResponseProperty("","");
+
+    if (includeName)
+        EspWUResponseProperty->setName(queryStatisticName(kind));
+    if (includeRawValue)
+    {
+        StringBuffer rawValue;
+        rawValue.append(value);
+        EspWUResponseProperty->setRawValue(rawValue);
+    }
+    SCMStringBuffer tmpStr;
+    if (includeFormatted)
+        EspWUResponseProperty->setFormatted(extra.getFormattedValue(tmpStr).str());
+    if (includeMeasure && extra.getMeasure()!=SMeasureNone)
+        EspWUResponseProperty->setMeasure(queryMeasureName(extra.getMeasure()));
+    if (includeCreator)
+        EspWUResponseProperty->setCreator(extra.getCreator(tmpStr).str());
+    if (includeCreatorType && extra.getCreatorType()!=SCTnone)
+        EspWUResponseProperty->setCreatorType(queryCreatorTypeName(extra.getCreatorType()));
+
+    EspWUResponseProperties.append(*EspWUResponseProperty.getClear());
+    if (extra.getTimestamp()>maxTimestamp)
+        maxTimestamp = extra.getTimestamp();
+}
+
+void WUDetailsVisitor::noteAttribute(WuAttr attr, const char * value)
+{
+    if (extraAttributesRequested)
+    {
+        // If the Attribute is not in the standard return statistic list,
+        // then check if it's in the additional attribute list
+        if ((returnAttributeSpecified && !returnAttributeList->test(attr-WANone)))
+        {
+            if (additionalAttribsForCurScope==nullptr ||
+                    additionalAttribsForCurScope->find(attr)==additionalAttribsForCurScope->end())
```

Comment:
trivial: indentation

Reply:
Don't know how you noticed that! I had to look a couple of times before seeing the extra spaces.